### PR TITLE
Fix unit tests that have broken since we stopped running them 2 years ago

### PR DESCRIPTION
**Pull request by dgoodwin**
[TRT-588](https://issues.redhat.com//browse/TRT-588)

This has been broken for two years since https://github.com/openshift/release/commit/d3551759477c8e007fb10b440d2827b09f72a924#diff-b2313b8cab6965347396e7671d80d67c69c8e0eee0b80f8e8a3b2a7ed9b35481

Several tests have broken since, or been committed in a never working state.

One test is removed as it now requires kube clients and doesn't seem worth the investment to mock out.

Tests on disruption and alert matching are changed to no longer work off the committed query_results.json files, which change every update and break every test.

- Fix TestGetClosestP95Value tests.
- Fix examples_test.go.
- Fix disruption matcher tests to not rely on changing data.
- Fix alert matcher tests to not rely on changing data.
- Fix TestDecodeProvider nil pointers.
- Drop a test that has grown too coupled to cluster config for a unit.
- Fix TestEventCountExtractor.
- Fix backend sampler check connection test.

### CODE REVIEW

File: pkg/monitor/backenddisruption/disruption_backend_sampler_test.go [hunk 1]
```diff
@@ -24,10 +24,10 @@ func TestBackendSampler_checkConnection(t *testing.T) {
...
 			w.WriteHeader(302)
 			w.Write([]byte("302-bad-response"))
-		//case req.URL.Path == "/302": // the client-auto-follows
-		//	w.Header().Set("Location", "http://google.com")
-		//	w.WriteHeader(302)
-		//	w.Write([]byte("302"))
+		case req.URL.Path == "/302": // the client-auto-follows
+			w.Header().Set("Location", "http://google.com")
+			w.WriteHeader(302)
+			w.Write([]byte("302"))
```
Comment (dgoodwin) [inline pkg/monitor/backenddisruption/disruption_backend_sampler_test.go:30]: This looks like it was temporarily commented out for some reason and then accidentally committed.

File: pkg/synthetictests/duplicated_events_test.go [hunk 2]
```diff
@@ -7,49 +7,11 @@ import (
...
-				t.Fatal(err)
-			}
-
-			actual := testDuplicatedEventForStableSystem(events, nil, "unit-test")
-			if !strings.Contains(actual[0].FailureOutput.Output, test.output) {
-				t.Error(spew.Sdump(actual[0]))
-			}
-		})
-	}
-}
```
Comment (dgoodwin) [inline pkg/synthetictests/duplicated_events_test.go:51]: This code is now trying to get etcd revisions from a live cluster, and does not feel worth the effort to mock out.

File: pkg/synthetictests/duplicated_events_test.go [hunk 3]
```diff
@@ -75,21 +37,18 @@ func TestEventCountExtractor(t *testing.T) {
 			times: 0,
 		},
 		{
-			name:  "pod eviction failure",
-			input: "reason/MalscheduledPod pod/router-default-84c89f5bf8-5rdcb pod/router-default-84c89f5bf8-bg9ql should be one per node, but all were placed on node/ip-10-0-172-166.ec2.internal; evicting pod/router-default-84c89f5bf8-5rdcb (79 times)",
-			times: 79,
+			name:    "pod eviction failure",
+			input:   "reason/MalscheduledPod pod/router-default-84c89f5bf8-5rdcb pod/router-default-84c89f5bf8-bg9ql should be one per node, but all were placed on node/ip-10-0-172-166.ec2.internal; evicting pod/router-default-84c89f5bf8-5rdcb (79 times)",
+			message: "reason/MalscheduledPod pod/router-default-84c89f5bf8-5rdcb pod/router-default-84c89f5bf8-bg9ql should be one per node, but all were placed on node/ip-10-0-172-166.ec2.internal; evicting pod/router-default-84c89f5bf8-5rdcb",
```
Comment (dgoodwin) [inline pkg/synthetictests/duplicated_events_test.go:42]: I checked, this one appeared broken from the very commit it was added.

**Comment by stbenjam** [comment]
verify failed, looks like the generated stuff needs to be updated

File: pkg/synthetictests/duplicated_events_test.go (hunk 2 above)
Comment (deads2k) [reply pkg/synthetictests/duplicated_events_test.go:51]: > This code is now trying to get etcd revisions from a live cluster, and does not feel worth the effort to mock out.

fine, next time it breaks we should write a new unit test

**Comment by dgoodwin** [comment]
bindata updated. something tells me it's unused though as the test change was working without it.

File: examples/image-streams/image-streams-centos7.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
-  "kind": "List",
+  "kind": "ImageStreamList",
```
Comment (stbenjam) [inline examples/image-streams/image-streams-centos7.json:2]: It looks like this change is making FIPS fail

https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/27460/pull-ci-openshift-origin-master-e2e-aws-ovn-fips/1578481788341194752

Comment (stbenjam) [reply examples/image-streams/image-streams-centos7.json:2]: I don't know anything about this, but maybe the apiVersion should be image.openshift.io/v1? I see examples of that being used on ImageStreams (not many examples of ImageStreamList).

**Pull request merged**